Configurable redcap csv import mapping

tkallady opened the pull request:
Resolves #620

ignatiusm commented on application/admin-client/src/components/CSVImport.tsx:
```diff
@@ -0,0 +1,83 @@
+        { label: StateTerritory.TAS, value: StateTerritory.TAS },
+        { label: StateTerritory.SA, value: StateTerritory.SA },
+        { label: StateTerritory.WA, value: StateTerritory.WA },
+        { label: StateTerritory.NT, value: StateTerritory.NT },
```
Add Preferred contact method?

ignatiusm commented on application/user-client/src/pages/Register.tsx:
```diff
@@ -85,13 +85,21 @@ export default function Register() {
             delete prefillProfile.password // Do not prefill password
+            function toISO(date: string) {
+              try {
+                const [d, m, y] = date.split('/')
```
Can we also split on `-` ? I tried with 1-09-1984 and this did not parse (as is expected with current code).

ignatiusm reviewed (changes requested):
Hi @tkallady :

That import field match library is great. A few questions:

1) Is it possible to go back to an earlier step? If you see something you want to change (e.g. matching col names), do you just have to start again? Would be better if you could click back to step 2 for example.

2) Preferred contact method is missing from the list of options (and next of kin etc). Shouldn't we offer the option to prefill all possible fields (except password and dependents)? I realise that parsing won't work with the test csv because of redcap's factor storage, but if people are coming from xlsx or other systems they might have well-formed data.
<img width="274" height="581" alt="image" src="https://github.com/user-attachments/assets/f0eae995-e8d6-470e-9d6b-46e8c978cbac" />

3) Is it possibly to adjust the text in the first step of the modal? "Data that we expect" is a bit unclear. I feel like we should explain what this step is doing: "Fields that we can import to pre-fill participant registration page" or something like that. Current wording might make people think that things are not possible if they don't have all the fields.

4) for date parsing can we also split on `-`?

5) For date parsing can either specify that we're expecting either ISO or day month year, or add an option so that the parser uses UsaDateToISO function instead?

Otherwise, everything works as expected (and I love getting warned about import errors ✅ ).

tkallady commented:
I have added functionality for 1,2,3. Regarding dates, the library we are using for the csv only has limited customisation. I've added regex validation so that only d/m/y is accepted. Given the rest of the app is geared towards Australian use (e.g. only Australian states in state dropdown), I'd suggest we create an issue for 'internationalisation' and include alternative date formats in that piece of work.

ignatiusm commented on application/admin-client/src/components/CSVImport.tsx:
```diff
@@ -0,0 +1,124 @@
+    fieldType: { type: 'input' },
+  },
+  {
+    label: 'Date of Birth (d/m/y)',
```
Tiny nit-pick: can you please change to `dd/mm/yyyy` or `DD/MM/YYYY` ? `y` implies a two-digit year in some contexts.
